### PR TITLE
Newsletters: add 288 (2024-02-07)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic entries @harding

### CODE REVIEW

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+- [Bitcoin Core #29189][] deprecates libconsensus.  Libconsensus was an
+  attempt to make Bitcoin Core's consensus logic usable in other
+  software.  However, the library hasn't seen any significant adoption
+  and it has become a burden on maintenance of Bitcoin Core.  The plan
+  is to "not migrate it to CMake and let it end with v27. Any remaining
+  use-cases could be handled in the future by libbitcoinkernel."
+
+- [Bitcoin Core #28956][] removes adjusted time from Bitcoin Core and
+  warns users if their computer's clock appears to out of sync with the
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-02-07-newsletter.md:257]:
```suggestion
  warns users if their computer's clock appears to be out of sync with the
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+  local node for at least a minute.  This can allow removing peers with
+  unstable connections.
+
+- [Core Lightning #7022][] removes `lnprototest` from Core Lightning's
+  testing infrastructure.  See [Newsletter #145][news145 lnproto] for
+  our description of them being added.
+
+- [Core Lightning #6936][] adds infrastructure to assist with
+  deprecating CLN features.  Features are now deprecated in code using a
+  functions that automatically disable those features by default based
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-02-07-newsletter.md:286]:
```suggestion
  deprecating CLN features.  Features are now deprecated in code using
  functions that automatically disable those features by default based
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+  code still exists.  This avoids an occasional problem where a CLN
+  feature would be reported as deprecated but still continue functioning
+  by default for a long time after it was planned for removal, possibly
+  leading users to continue depending on it and making actual removal
+  more difficult.
+
+- [LND #8345][] begins testing whether transactions are likely to relay
+  before broadcasting them by calling a full node's `testmempoolaccept`
+  RPC when available.  This allows the node to detect potential problems
+  with the transaction before anything is sent to a third-party,
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-02-07-newsletter.md:298]:
```suggestion
  with the transaction before anything is sent to a third party,
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+  vulnerable to exploitation of the bug.  Upgrading to Bitcoin Core 22
+  or higher is strongly recommended.  To the best of our knowledge, no
+  one has lost funds due to the attack described below.
+
+    An attacker finds an LN forwarding node that is associated with a
+    relaying Bitcoin node running a version of Bitcoin Core earlier
+    than 22.  The attacker opens many separate connections to a victim's
+    Bitcoin node.  The attacker then attempts to deliver newly found
+    blocks to the victim faster than any honest peers, resulting in
+    victim's node automatically assigning peers controlled by the
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:30]:
```suggestion
    the victim's node automatically assigning peers controlled by the
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+    attacker to all of the victim's high-bandwidth [compact block
+    relay][topic compact block relay] slots.
+
+    After the attacker obtains control over many of the victim's Bitcoin
+    peer slots, it uses channels it controls on either side of the
+    victim to forward payments it creates.  For example:
+
+      Attacker Spender -> Victim Forwarder -> Attacker Receiver
+
+    The attacker works with a miner create a block that unilaterally
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:40]:
```suggestion
    The attacker works with a miner to create a block that unilaterally
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+    spending node is able to receive a refund of the payment.  If that
+    happens, the attacker's spending node paid nothing and the
+    attacker's receiving node received an amount extracted from the
+    victim's node.
+
+    As Siegel reports, two changes were made to Bitcoin Core to prevent
+    stalling:
+
+    - [Bitcoin Core #22144][] randomizes the order in which peers are
+      serviced in the message handling thread.  See [Newsletter
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:72]:
```suggestion
      serviced in the message-handling thread.  See [Newsletter
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+      #154][news154 stall].
+
+    - [Bitcoin Core #22147][] keeps at least one outbound high bandwidth
+      compact block peer even if inbound peers seem to be performing
+      better.  The local node selects its own outbound peers, meaning
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:77]:
```suggestion
      better.  The local node selects its outbound peers, meaning
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+  channel open transaction receives a sufficient number of
+  confirmations, so there’s no risk to the acceptor spending some of
+  those funds back through the funder using the standard LN protocol.
+  The initial proposal for v3 transaction relay policy would only allow
+  an unconfirmed v3 transaction to have, at most, a single child in the
+  mempool; the expectation is that the single child will [CPFP fee
+  bump][topic cpfp] its parent if necessary.
+
+    Those v3 rules are incompatible with both parties being able to fee
+    bump a zero-conf channel open: the funding transaction which creates
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:97]:
```suggestion
    bump a zero-conf channel open: the funding transaction that creates
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+    the channel is the parent of a v3 transaction which closes the
+    channel and the grandparent of a v3 transaction for fee bumping.
+    Since the v3 rules only allow one parent and one child, there's no
+    way for the funding transaction to be fee bumped without modifying
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:101]:
```suggestion
    way for the funding transaction to be fee-bumped without modifying
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+    how it is created.  Bastien Teinturier [notes][teinturier splice]
+    that [splicing][topic splicing] encounters a similar problem.
+
+    As of this writing, the main solution proposed appeared to be
+    modifying funding and splicing transactions to include an extra
+    output for CPFP fee bumping now, wait for [cluster mempool][topic
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:107]:
```suggestion
    output for CPFP fee bumping now, waiting for [cluster mempool][topic
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+    cluster mempool] to hopefully allow v3 to permit more permissive
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:108]:
```suggestion
    cluster mempool] to allow v3 to permit more permissive
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,319 @@
...
+    spending a segwit v0 output is for her to obtain a copy of the
+    entire previous transaction that contained Bob's output.  If Alice
+    doesn't perform this check, Bob can lie about spending a segwit
+    output and instead spend a legacy output that allows him to mutate
+    the txid, allowing him to invalidate the refund transaction and refuse
+    to return any funds to Alice unless she agrees to pay him a ransom.
+
+    For segwit v1, each `SIGHASH_ALL` signature directly commits to
+    every previous output being spent in the transaction (see
+    [Newsletter #97][news97 spk]), so Alice can require Bob disclose his
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:139]:
```suggestion
    [Newsletter #97][news97 spk]), so Alice can require Bob to disclose his
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 13]
```diff
@@ -0,0 +1,319 @@
...
+    signature commit to it.  Now, if Bob lied and actually had a
+    non-segwit output, the commitment made by Alice's signature wouldn't
+    be valid, so the signature wouldn't be valid, the funding
+    transaction wouldn't confirm, and there would be no need for a
+    refund.
+
+    This leads to two rules that protocols depending on presigned
+    refunds must follow for security:
+
+    1. If you are contributing a input, prefer to contribute an input
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:152]:
```suggestion
    1. If you are contributing an input, prefer to contribute an input
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 14]
```diff
@@ -0,0 +1,319 @@
...
+  peers, allowing it to avoid unnecessarily rejecting blocks and also
+  give blocks it produced a more accurate time.  However, adjusted time
+  has also led to problems in the past and does not provide meaningful
+  benefits to nodes on the current network.
+
+- [Bitcoin Core #29347][] enables [v2 P2P transport][topic v2 p2p
+  transport] by default.  New connections between two peers that both
+  support the v2 protocol will use encryption.
+
+- [Core Lightning #6985][] adds options to `hsmtool` that allow it to
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:270]:
```suggestion
- [Core Lightning #6985][] adds options to `hsmtool` that allows it to
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 15]
```diff
@@ -0,0 +1,319 @@
...
+  return the private keys for the onchain wallet in a way that allows
+  those keys to imported into another wallet.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:272]:
```suggestion
  those keys to be imported into another wallet.
```

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 16]
```diff
@@ -0,0 +1,319 @@
...
+  testing infrastructure.  See [Newsletter #145][news145 lnproto] for
+  our description of them being added.
+
+- [Core Lightning #6936][] adds infrastructure to assist with
+  deprecating CLN features.  Features are now deprecated in code using a
+  functions that automatically disable those features by default based
+  on the current program version.  Users can still force enable the
+  features even after their indicated deprecation version as long as the
+  code still exists.  This avoids an occasional problem where a CLN
+  feature would be reported as deprecated but still continue functioning
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:290]:
```suggestion
  feature would be reported as deprecated but continues functioning
```

File: _posts/en/newsletters/2024-02-07-newsletter.md (hunk 11 above)
Comment (harding) [reply _posts/en/newsletters/2024-02-07-newsletter.md:108]: Leaving as-is. I don't think we can say for certain that cluster mempool will make that possible.

File: _posts/en/newsletters/2024-02-07-newsletter.md (hunk 14 above)
Comment (harding) [reply _posts/en/newsletters/2024-02-07-newsletter.md:270]: Leaving as-is.  I think the singular here agrees with the plural "options" earlier in the sentence.

**Comment by harding** [comment]
Made edits or left a reply on all feedback so far.  Thanks @vostrnad and @LarryRuane !

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 17]
```diff
@@ -94,17 +94,17 @@ This week's newsletter FIXME
...
+    bump a zero-conf channel open: the funding transaction that creates
     the channel is the parent of a v3 transaction which closes the
     channel and the grandparent of a v3 transaction for fee bumping.
     Since the v3 rules only allow one parent and one child, there's no
-    way for the funding transaction to be fee bumped without modifying
+    way for the funding transaction to be fee-bumped without modifying
     how it is created.  Bastien Teinturier [notes][teinturier splice]
     that [splicing][topic splicing] encounters a similar problem.
 
     As of this writing, the main solution proposed appeared to be
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-07-newsletter.md:105]:
```suggestion
    As of this writing, the main solution proposed appears to be
```

File: _posts/en/newsletters/2024-02-07-newsletter.md (hunk 11 above)
Comment (LarryRuane) [reply _posts/en/newsletters/2024-02-07-newsletter.md:108]: That's fine, I think readers will understand what is meant, but I remember reading a long time ago that "hopefully" is often used incorrectly, and shouldn't be used at all in many cases. I can't recall the details, but I think the problem is (using this as an example), "hopefully" appears to be modifying "allow", which is not the intended meaning. To me, it reads as imprecise language (but maybe it's just me). Maybe something like:
```
As of this writing, the main solution proposed appeared to be
modifying funding and splicing transactions to include an extra
output for CPFP fee bumping now, in anticipation of cluster mempool
allowing v3 to permit more permissive topologies ...
```
(I think it's understood that anticipated events might never occur. -- especially in bitcoin!)

File: _posts/en/newsletters/2024-02-07-newsletter.md (hunk 14 above)
Comment (LarryRuane) [reply _posts/en/newsletters/2024-02-07-newsletter.md:270]: The "it" here refers to hsmtool (singular), not options. But no big deal.

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 18]
```diff
@@ -0,0 +1,319 @@
...
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Bitcoin Core #29189][] deprecates libconsensus.  Libconsensus was an
+  attempt to make Bitcoin Core's consensus logic usable in other
+  software.  However, the library hasn't seen any significant adoption
+  and it has become a burden on maintenance of Bitcoin Core.  The plan
+  is to "not migrate it to CMake and let it end with v27. Any remaining
+  use-cases could be handled in the future by libbitcoinkernel."
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-07-newsletter.md:254]: Perhaps libbitcoinkernel can link here: https://github.com/bitcoin/bitcoin/issues/27587

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 19]
```diff
@@ -0,0 +1,319 @@
...
+
+- [Bitcoin Core #28956][] removes adjusted time from Bitcoin Core and
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-07-newsletter.md:256]: Could link adjusted time to https://bitcoinops.org/en/newsletters/2024/01/10/#bitcoin-core-pr-review-club

File: _posts/en/newsletters/2024-02-07-newsletter.md [hunk 20]
```diff
@@ -0,0 +1,319 @@
...
+  unstable connections.
+
+- [Core Lightning #7022][] removes `lnprototest` from Core Lightning's
+  testing infrastructure.  See [Newsletter #145][news145 lnproto] for
+  our description of them being added.
+
+- [Core Lightning #6936][] adds infrastructure to assist with
+  deprecating CLN features.  Features are now deprecated in code using
+  functions that automatically disable those features by default based
+  on the current program version.  Users can still force enable the
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-07-newsletter.md:287]: Maybe 
```suggestion
  on the current program version.  Users can still force-enable the
```

**Review by bitschmidty — COMMENTED** [review]
Sorry for the late review comments!

**Comment by harding** [comment]
Made edits for second round of feedback (thanks @LarryRuane @bitschmidty !), added lede, releases/RCs, and topic entries.  Thanks everyone!

**Comment by bitschmidty** [comment]
Squashed and merged, thank you @harding @vostrnad @LarryRuane

**Pull request merged**